lib/xmpp-client.js

```diff
@@ -0,0 +1,42 @@
+import xmpp from 'xmpp.js';
+
+const domain = 'im.wordpress.com';
+
+export const listenForSMS = ( user ) => {
+	const jabberId = user.username + '@' + domain;
+	const password = user.applicationPassword;
+	const client = new xmpp.Client( { jid: jabberId } );
+
+	client.plugin( require( '@xmpp/plugins/starttls' ) );
+	client.plugin( require( '@xmpp/plugins/session-establishment' ) );
+
+	client.start( { uri: 'xmpp://xmpp.wordpress.com', domain: domain } );
+
+	client.on( 'stanza', stanza => {
+		if ( ! stanza.is( 'message' ) || stanza.getAttr( 'from' ) !== 'wpcom@im.wordpress.com' ) {
+			return;
+		}
+		const body = stanza.getChild( 'body' );
+		if ( ! body ) {
+			return;
+		}
+		const messageText = body.getText();
+		const smsContentRegex = /^SMS received from (\+?\d+): (.*)$/.exec( messageText );
+		if ( ! smsContentRegex || ! smsContentRegex[2] ) {
```

Comment:
should we check that `smsContentRegex.length > 2` here instead directly accessing 3rd index ?

Reply:
👍 I negated this condition (to return early) without thinking.